examples/mnist/mnist_softmax_distributed.py

```diff
@@ -0,0 +1,126 @@
+    b = tf.Variable(tf.zeros([10]))
+    y = tf.matmul(x, W) + b
+
+  # Define loss and optimizer
+    y_ = tf.placeholder(tf.float32, [None, 10])
+
+  # The raw formulation of cross-entropy,
+  #
+  #   tf.reduce_mean(-tf.reduce_sum(y_ * tf.log(tf.nn.softmax(y)),
+  #                                 reduction_indices=[1]))
+  #
+  # can be numerically unstable.
+  #
+  # So here we use tf.nn.softmax_cross_entropy_with_logits on the raw
+  # outputs of 'y', and then average across the batch.
+  cross_entropy = tf.reduce_mean(
+      tf.nn.softmax_cross_entropy_with_logits(labels=y_, logits=y))
+  #global_step = tf.train.get_or_create_global_step()
+  global_step = tf.contrib.framework.get_or_create_global_step()
+  opt = tf.train.GradientDescentOptimizer(0.5)
+  # Add MPI Distributed Optimizer
+  #import pdb; pdb.set_trace()
+  with tf.name_scope("horovod_opt"):
+    opt = hvd.DistributedOptimizer(opt) 
+  train_step = opt.minimize(cross_entropy, global_step=global_step)
+
+  # The StopAtStepHook handles stopping after running given steps. 
+  hooks=[
+    hvd.BroadcastGlobalVariablesHook(0),
+    tf.train.StopAtStepHook(last_step=10)]
+
+  # Test trained model
+  correct_prediction = tf.equal(tf.argmax(y, 1), tf.argmax(y_, 1)) 
+  accuracy = tf.reduce_mean(tf.cast(correct_prediction, tf.float32)) 
+
+  # Enable soft placement and tracing as needed
+  config = tf.ConfigProto(
+    allow_soft_placement=True, 
+    log_device_placement=True,
+    inter_op_parallelism_threads=1)
+
+  #config.graph_options.optimizer_options.global_jit_level = jit_level
+  run_metadata = tf.RunMetadata()
+
+  #init_op = tf.global_variables_initializer()
+  print("Variables initialized ...")
+
+  # The MonitoredTrainingSession takes care of session initialization
+  with tf.train.MonitoredTrainingSession(hooks=hooks,
+                                         config=config) as mon_sess:
+    start = time.time()
+    train_writer = tf.summary.FileWriter(FLAGS.log_dir, mon_sess.graph)
+    while not mon_sess.should_stop():  
+      # Train
+      batch_xs, batch_ys = mnist.train.next_batch(100)
+      mon_sess.run(train_step, feed_dict={x: batch_xs, y_: batch_ys})
+      
+      # Test trained model
+#      if not mon_sess.should_stop():
+#        print("Accuracy: ", mon_sess.run(accuracy, feed_dict={x: mnist.test.images,
+#                                            y_: mnist.test.labels}))
+
+    end = time.time()
+
+  if hvd.rank() == 0:
+    print("Training time: %f seconds" % (end-start))
+
+if __name__ == '__main__':
+  parser = argparse.ArgumentParser()
+  parser.add_argument('--data_dir', type=str, default='/tmp/tensorflow/mnist/input_data',
+                      help='Directory for storing input data')
+  parser.add_argument('--log_dir',type=str, default='/tmp/tensorflow/mnist/logs/mnist_with_summaries',
+                      help='Summaries log directory')
+  FLAGS, unparsed = parser.parse_known_args()
+  tf.app.run(main=main, argv=[sys.argv[0]] + unparsed)
```

Comment:
Can you please add the command line to run this as a comment in the end of the file

Reply:
Added an example run command for distributed runs